conan/tools/cmake/toolchain/blocks.py

```diff
@@ -667,6 +667,22 @@ class TryCompileBlock(Block):
         """)
 
 
+class CompilersBlock(Block):
+    template = textwrap.dedent(r"""
+        {% for lang, compiler_path in compilers.items() %}
+        set(CMAKE_{{ lang }}_COMPILER "{{ compiler_path|replace('\\', '/') }}")
+        {% endfor %}
+    """)
+
+    def context(self):
+        # Reading configuration from "tools.build:compilers" -> {"C": "/usr/bin/gcc"}
+        compilers = self._conanfile.conf.get("tools.build:compilers", default={}, check_type=dict)
+        # Users used to define CC as env variable. Translating it just in case.
+        if "CC" in compilers:
+            compilers["C"] = compilers.pop("CC")
+        return {"compilers": compilers}
```

Comment:
I'm not sure about this... if we are directly translating the dictionary key to the CMAKE_<key>_COMPILER CMake variable, then we know that are a more or less restricted set of accepted keys, something like: `CXX`, `C`, `CSharp`, `CUDA`, `Fortran`, `HIP`, `ISPC`, `ASM`... that are possible values for the <LANG> in CMake. Maybe we should fail in this case? Maybe we should define the set of generic keys we accept that are more generic?